sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3286,101 +3283,91 @@ pi_result piQueueGetInfo(pi_queue Queue, pi_queue_info ParamName,
 }
 
 pi_result piQueueRetain(pi_queue Queue) {
-  // Lock automatically releases when this goes out of scope.
-  std::scoped_lock lock(Queue->Mutex);
   Queue->RefCountExternal++;
-  piQueueRetainNoLock(Queue);
+  piQueueRetainInternal(Queue);
```

Comment:
Do you know why we have to counters? Any chance that it's because we assumed we needed a whole lock for one but not for another? If so, we might be able to just merge them in this PR.

Reply:
If I don't miss anything, we can't merge these ref counts. External ref count equal to 0 means that there are no external references anymore and synchronize the queue and reset command lists in this queue. It is possible that at this point internal ref count is still not zero. After all internal references to this queue disappear, i.e. internal ref count = 0 then we can delete the queue.
Shortly, external ref count is used to detect the point when we can synchronize and internal ref count is used to detect the point when we can delete.